### PR TITLE
Malaysian Localization Provider

### PR DESCRIPTION
**Pull request by arifulhb**
Including
- Address
- Internet
- Person
- Payment
- PhoneNumber

And TestCases

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1030,6 +1030,33 @@ echo $faker->borough; // "강남구"
...
+<?php
+
+// Prefix
+echo $faker->prefix; // "Y.Bhg"
+
+// Second Personal Name Male
+echo $faker->secondPersonalNameMale; // "Mohammad"
+
+// Second Personal Name Female
+echo $faker->secondPersonalNameMale; // "Siti"
```
Comment (fzaninotto) [inline readme.md]: bad copy/paste

File: src/Faker/Generator.php [hunk 2]
```diff
@@ -11,6 +11,8 @@
  * @property string $title
  * @property string $titleMale
  * @property string $titleFemale
+ * @property string $secondPersonalNameMale
```
Comment (fzaninotto) [inline src/Faker/Generator.php]: no, this is only for generally-available formatters, not for locale-specific ones.

File: src/Faker/Provider/ms_MY/Address.php [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        'Sarawak',
+        'Selangor',
+        'Terengganu'
+    );
+
+    /**
+     * @link https://en.wikipedia.org/wiki/List_of_cities_in_Malaysia
+     */
+    protected static $cityNames = array(
+        'George Town',
```
Comment (fzaninotto) [inline src/Faker/Provider/ms_MY/Address.php]: Can you order it alphabetically?

File: src/Faker/Provider/ms_MY/Address.php [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+    public function streetName()
+    {
+        $format = static::randomElement(static::$streetNames);
+
+        return $this->generator->parse($format);
+    }
+
+    protected static $country = array(
+        'Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra', 'Angola', 'Anguilla', 'Antarctica (the territory South of 60 deg S)', 'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba', 'Australia', 'Austria', 'Azerbaijan',
```
Comment (fzaninotto) [inline src/Faker/Provider/ms_MY/Address.php]: This is English, not malaisian

File: src/Faker/Provider/ms_MY/Person.php [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    /**
+     * @link https://en.wikipedia.org/wiki/List_of_Malaysians
+     */
+    protected static $firstNameFemale = array(
+        'Shila', 'Yasmin', 'Kasma', 'Erra', 'Fazira', 'Fauziah', 'Sheila', 'Amy', 'Nurhaliza', 'Misha', 'Fasha Sandha',
+        'Hannah', 'Scha', 'Adira', 'Aini', 'Yuna', 'Nurhaliza', 'Sophia', 'Damia', 'Zara', 'Aishah', 'Alya', 'Humaira',
+        'Farah', 'Sofea', 'Mira', 'Wani', 'Husna' ,'Nadia', 'Insyirah', 'Sasha', 'Sabrina', 'Aziah', 'Aqila'
+    );
+
+    protected static $lastName = array(
```
Comment (fzaninotto) [inline src/Faker/Provider/ms_MY/Person.php]: please provide a source

File: src/Faker/Provider/ms_MY/Person.php [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        '{{prefix}} {{secondPersonalNameFemale}} {{firstNameFemale}} binti {{lastName}}',
+        '{{prefix}} {{firstNameFemale}} binte {{lastName}}',
+        '{{prefix}} {{firstNameFemale}} binti {{lastName}}',
+    );
+
+    /**
+     * @link https://en.wikipedia.org/wiki/List_of_Malaysians
+     */
+    protected static $firstNameMale = array(
+        'Sudirman', 'Awal', 'Asmawi', 'Azhar', 'Sean', 'Aziddin', 'Shanon', 'Hani', 'Mohsin', 'Aziz', 'Azmyl', 'Aznil',
```
Comment (fzaninotto) [inline src/Faker/Provider/ms_MY/Person.php]: In this dataset, as well as in the other ones of this class, there is not enough data to make a good randomization. Can't you find more?

Comment (arifulhb) [inline src/Faker/Provider/ms_MY/Person.php]: I will add more data here.

File: src/Faker/Generator.php (hunk 2 above)
Comment (arifulhb) [inline src/Faker/Generator.php]: ohh ok.. will remove it.

File: src/Faker/Provider/ms_MY/Address.php (hunk 3 above)
Comment (arifulhb) [inline src/Faker/Provider/ms_MY/Address.php]: Sure.. coming in next commit.

**Comment by arifulhb** [comment]
**Person**

- More names added
- Links added

**Address**

- Country list is updated as Bahasa Malayu
- City and State Names are sorted alphabetically

**Other**

- readme.md updated
- Property removed from `Generator.php`

**Comment by fzaninotto** [comment]
Thanks, can you add more data for first and last name?

**Comment by arifulhb** [comment]
Added few more names as much I've found with reference.

**Comment by fzaninotto** [comment]
please order the lists alphabetically

File: src/Faker/Provider/ms_MY/Person.php [hunk 7]
```diff
@@ -43,13 +43,16 @@ class Person extends \Faker\Provider\Person
 
     /**
      * @link https://en.wikipedia.org/wiki/List_of_Malaysians
+     * @link http://www.studentsoftheworld.info/penpals/stats.php3?Pays=MLS
```
Comment (fzaninotto) [inline src/Faker/Provider/ms_MY/Person.php]: this data isn't free:

> (c) Students of the World / Etudiants du Monde - www.StudentsOfTheWorld.info

**Comment by fzaninotto** [comment]
This pr is stale, closing.

**Pull request closed without merging**